### PR TITLE
Merging view : HvdcLine adapter implementation

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
New adaptated classes are implemented.
Each method adaptated not implemented yet throw an exception

**What is the current behavior?** *(You can also link to an open issue here)*
Merging view module implementation need to implement hvdcline adapter

**What is the new behavior (if this is a feature change)?**
Merging view module implementation step : hvdcline adapter implementation needed
HvdcLine creation are not allowed between two merged network, exception will be thrown

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import com.powsybl.iidm.network.HvdcLineAdder;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.util.Identifiables;
+
+import java.util.Objects;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class HvdcLineAdderAdapter implements HvdcLineAdder {
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java]: Why the design of this adder is completely different from the others: why don't we have a delegate here?

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        checkAndSetUniqueId();
+        return index.getHvdcLine(addLine(n1));
+    }
+
+    private Network checkAndGetNetwork1() {
+        if (Objects.isNull(converterStationId1)) {
+            throw new PowsyblException("Side 1 converter station is not set");
+        }
+        final Network network = index.getNetworkStream()
+                                     .filter(n -> n.getHvdcConverterStation(converterStationId1) != null)
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java]: The merging view should implement `getHvdcConverterStation(String)`: why don't you use it?

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    private Network checkAndGetNetwork2() {
+        if (Objects.isNull(converterStationId2)) {
+            throw new PowsyblException("Side 2 converter station is not set");
+        }
+        final Network network = index.getNetworkStream()
+                                     .filter(n -> n.getHvdcConverterStation(converterStationId2) != null)
+                                     .findFirst()
+                                     .orElse(null);
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java]: Same remark here

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelAdapter.java [hunk 4]
```diff
@@ -511,7 +511,22 @@ public StaticVarCompensatorAdder newStaticVarCompensator() {
...
+    }
+
+    @Override
+    public LccConverterStationAdder newLccConverterStation() {
+        return new LccConverterStationAdderAdapter(getDelegate().newLccConverterStation(), getIndex());
+    }
+
+    @Override
+    public Iterable<LccConverterStation> getLccConverterStations() {
+        return getLccConverterStationStream().collect(Collectors.toSet());
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelAdapter.java]: Is the set needed? Is the order important?

Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelAdapter.java]: My mistake, i willl change it to List

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java (hunk 2 above)
Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java]: We need to search which network contains "converterStationId1" & "converterStationId1"
In order to know, if we are creating HvdcLine between 2 networks or not.
For now, no creation are allowed between 2 networks
So, i need to get reference on networks contains into mergingview

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java (hunk 1 above)
Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java]: I think, the delegate cannot be used here
Which network will be used to create the HvdcLine ?
The network to use cannot be known before `setConverterStationId` methods

Comment (mathbagu) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/HvdcLineAdderAdapter.java]: Right... My bad, it miss something when I did the review

**Pull request merged**